### PR TITLE
Gtw/prepare_for_usage

### PR DESCRIPTION
**Pull request by gregertw**
* Refactor all config files into defaults and local versions
* Add demo pre-processing to deployment

### CODE REVIEW

File: .github/workflows/dev.yml [hunk 1]
```diff
@@ -25,6 +25,9 @@ jobs:
           virtualenvs-create: false
       - name: Install dependencies
         run: poetry install
+      - name: "Pre-processing for demo environment"
+        run: |
+          ./demo/preproc.sh
```
Comment (doctrino) [inline .github/workflows/dev.yml:30]: Any reason to introduce bash and not just stick with Python?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -53,19 +53,20 @@ The configuration files are loaded directly into the Python SDK's support data c
 use towards the CDF API. No client side schema validation should be done to ensure that you can immediately
 add a yaml configuration property without upcoming anything else than the version of the Python SDK.
 
+> NOTE!! As of now, any non-recognised properties will just be ignored by the Python SDK. If you don't
+> get the desired configuration deployed, check your spelling and use of snake_case vs camelCase.
```
Comment (doctrino) [inline CONTRIBUTING.md:57]:
```suggestion
> NOTE!! As of now, any non-recognised properties will just be ignored by the Python SDK. If you don't
> get the desired configuration deployed, check your spelling and use of snake_case vs camelCase. The Python SDK
> expects camelCase.
```

File: README.md [hunk 3]
```diff
@@ -27,17 +27,19 @@ The templates have been designed to be copied and checked into your own
...
 using YAML configuration files. The most important files are [local.yaml](./local.yaml) where you specify what
 to load into a project, and the root [config.yaml](./config.yaml) file where you can set global
-variables.
+variables. You will find globally set configuration defaults in `default.config.yaml` files. These variables
+can always be overridden by setting the same variable in a `config.yaml` file in the same directory.
 
 ### Modules
 
-The basic concepts are **modules** and **packages**. Modules live in the `modules/`, `common/`, and `examples/`
+The basic concepts are **modules** and **packages**. Modules live in the `modules/`, `local_modules`, `common/`, and `examples/`
```
Comment (doctrino) [inline README.md:35]: I like this, good to have a preferred way of adding your own stuff.

File: scripts/templates.py [hunk 4]
```diff
@@ -84,23 +86,41 @@ def read_environ_config(
...
     """
 
+    files = []
+    if name is None:
+        # Order is important!
+        for directory in yaml_dirs:
+            for yaml_file in Path(directory).glob("default.config.yaml"):
+                files.append(yaml_file)
+            for yaml_file in Path(directory).glob("config.yaml"):
+                files.append(yaml_file)
```
Comment (doctrino) [inline scripts/templates.py:108]:
```suggestion
        for directory in yaml_dirs:
            files.extend(Path(directory).glob("default.config.yaml"))
            files.extend(Path(directory).glob("config.yaml"))
```
I think that is the same.

File: scripts/templates.py [hunk 5]
```diff
@@ -84,23 +86,41 @@ def read_environ_config(
...
-                config_data = yaml.safe_load(yaml_file.read_text())
-            except yaml.YAMLError:
-                print(f"Error reading {yaml_file}")
-                continue
-            data.update(config_data)
+    for yaml_file in files:
+        try:
+            config_data = yaml.safe_load(yaml_file.read_text())
+        except yaml.YAMLError:
+            print(f"Error reading {yaml_file}")
```
Comment (doctrino) [inline scripts/templates.py:121]:
```suggestion
        except yaml.YAMLError as e:
            print(f"Error reading {yaml_file}: {e}")
```

**Review by doctrino — APPROVED** [review]
Just some minor stuff.

File: .github/workflows/dev.yml (hunk 1 above)
Comment (gregertw) [reply .github/workflows/dev.yml:30]: Not beyond "simplest tool for the job" (two cp commands)

**Pull request merged**